packages/jest-cli/src/cli/index.js

```diff
@@ -137,8 +137,11 @@ const readResultsAndExit = (
   globalConfig: GlobalConfig,
 ) => {
   const code = !result || result.success ? 0 : globalConfig.testFailureExitCode;
-
-  process.on('exit', () => (process.exitCode = code));
+  
+  // Only exit if needed
+  if (code) {
```

Comment:
Can you be explicit about `!== 0`?

Reply:
So I thought about that, but the problem is, process.exitCode is undefined. I could check for not zero and not undefined though?

Reply:
Will do!